study1/week1/Binary_Search_Tree.java

```diff
@@ -0,0 +1,20 @@
+public class Binary_Search_Tree {
+    // using dynamic programming
+    public int numTrees(int n) {
+        int dp[] = new int[n + 1]; // size of the dp array is n + 1, space complexity: O(n+1)
```

Comment:
I guess you already know, but We drop all the constants from the big O notation
so `O(n)`